src/components/ReservationModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import styled from 'styled-components';
+import { WHITE, BLACK } from '../colors';
+import { Modal } from 'antd';
+
+interface ReservationModalProps {
+    readonly status: string;
+    readonly blockID: number;
+}
```

Comment:
You should add three properties here: 
```
readonly onOk: () => void;
readonly onCancel: () => void;
readonly isVisible: boolean;
```
That way we can respond to the user's response with custom code. 
And we'll control visibility from the parent component.